### PR TITLE
Optimize block creation in the tests

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Hopefully it speeds up the tests a bit.

### CODE REVIEW

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -955,7 +955,9 @@ public class TestAPIManager
         static assert (isInputRange!Idxs);
 
         // Get the last block from the first client
-        auto lastBlock = this.clients[client_idxs.front].getAllBlocks()[$ -1];
+        auto client = this.clients[client_idxs.front];
+        const last_height = client.getBlockHeight();
+        auto lastBlock = client.getBlock(last_height);
```
Comment (linked0) [inline source/agora/test/Base.d]: `auto lastBlock = client.getBlock(client.getBlockHeight());`

Suggested change.

File: source/agora/test/Base.d [hunk 2]
```diff
@@ -986,24 +988,22 @@ public class TestAPIManager
         static assert (isInputRange!Idxs);
 
         auto first_client = this.clients[client_idxs.front];
-        auto all_blocks = first_client.getAllBlocks();
+        const last_height = first_client.getBlockHeight();
+        const last_block = first_client.getBlock(last_height);
```
Comment (linked0) [inline source/agora/test/Base.d]: Ditto

**Review by linked0 — APPROVED** [review]
LGTM

**Review by kchulj — APPROVED** [review]
LGTM.

**Review by Geod24 — APPROVED** [review]
LGTM. Just rebased on master to test something, feel free to merge when green.

**Pull request merged**